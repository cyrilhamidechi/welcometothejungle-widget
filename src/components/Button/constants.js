export const COLORS = ["green", "grey"];
