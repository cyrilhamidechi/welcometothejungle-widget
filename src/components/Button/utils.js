import { colors } from "../../config";

export const getBackgroundHover = ({ color }) => {
  switch (color) {
    case "grey":
      return colors.greyLight;
    default:
      return colors.greenLight;
  }
};
