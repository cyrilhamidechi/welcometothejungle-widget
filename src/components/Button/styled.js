import styled from "styled-components";

import { colors } from "../../config";
import { getBackgroundHover } from "./utils";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${({ onlyIcon }) => onlyIcon && "30px"};
  height: 30px;
  padding: ${({ onlyIcon }) => (onlyIcon ? "0" : "0 20px")};
  border: 0;
  border-radius: ${({ onlyIcon }) => (onlyIcon ? "30px" : "4px")};
  background: ${({ color }) => colors[color]};
  color: ${colors.white};
  font-size: 14px;
  text-decoration: none;
  cursor: pointer;
  transition: background 300ms ease-in-out, opacity 300ms;

  &:hover:enabled {
    background: ${getBackgroundHover};
  }

  &:disabled {
    opacity: 0.3;
  }
`;

export const Text = styled.span`
  display: flex;
  align-items: center;
  margin-left: "10px";
`;
