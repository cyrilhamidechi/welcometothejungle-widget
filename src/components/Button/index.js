import PropTypes from "prop-types";
import React, { PureComponent } from "react";

import Icon from "./../../components/Icon";
import { COLORS } from "./constants";

import * as S from "./styled";

export default class Button extends PureComponent {
  static propTypes = {
    blank: PropTypes.bool,
    children: PropTypes.node,
    color: PropTypes.oneOf(COLORS),
    href: PropTypes.string,
    iconName: PropTypes.string,
    onClick: PropTypes.func
  };

  static defaultProps = {
    color: "green"
  };

  handleClick = () => {
    const { onClick } = this.props;

    onClick && onClick();
  };

  render() {
    const {
      blank,
      children,
      color,
      disabled,
      iconName,
      href,
      size
    } = this.props;

    // get props for link
    const linkProps = {
      as: "a",
      href,
      target: blank ? "_blank" : undefined,
      rel: blank ? "noopenner noreferrer" : undefined
    };

    // get props for button
    const buttonProps = {
      as: "button",
      onClick: this.handleClick,
      disabled
    };

    // if href exist it's a link else it's a button
    const getProps = href ? linkProps : buttonProps;

    // check if had only an icon
    const onlyIcon = iconName && !children;

    return (
      <S.Wrapper {...getProps} color={color} onlyIcon={onlyIcon} size={size}>
        {iconName && <Icon name={iconName} size="20" color="white" />}
        {iconName ? <S.Text>{children}</S.Text> : children}
      </S.Wrapper>
    );
  }
}
