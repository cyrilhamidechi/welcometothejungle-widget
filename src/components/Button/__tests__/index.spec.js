import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-styled-components";

import Button from "./../";

describe("Button", () => {
  describe("Link", () => {
    it("should render the link button", () => {
      const tree = renderer
        .create(
          <Button href="http://www.test.com">
            <div>Click me</div>
          </Button>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render the link button with blank", () => {
      const tree = renderer
        .create(
          <Button href="http://www.test.com" blank>
            <div>Click me</div>
          </Button>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("Button", () => {
    it("should render the button", () => {
      const onClick = jest.fn();
      const tree = renderer
        .create(
          <Button onClick={onClick}>
            <div>Click me</div>
          </Button>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should call onClick", () => {
      const onClick = jest.fn();
      const tree = shallow(
        <Button onClick={onClick}>
          <div>Click me</div>
        </Button>
      );

      tree.simulate("click");

      expect(onClick).toHaveBeenCalled();
    });
  });
});
