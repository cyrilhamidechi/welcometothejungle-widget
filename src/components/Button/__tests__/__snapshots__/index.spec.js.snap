// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button should render the button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 20px;
  border: 0;
  border-radius: 4px;
  background: #00c29a;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
}

.c0:hover {
  background: #37cbaa;
}

<button
  className="c0"
  color="green"
  onClick={[Function]}
>
  <div>
    Click me
  </div>
</button>
`;

exports[`Button Link should render the link button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 20px;
  border: 0;
  border-radius: 4px;
  background: #00c29a;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
}

.c0:hover {
  background: #37cbaa;
}

<a
  className="c0"
  color="green"
  href="http://www.test.com"
>
  <div>
    Click me
  </div>
</a>
`;

exports[`Button Link should render the link button with blank 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 20px;
  border: 0;
  border-radius: 4px;
  background: #00c29a;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
}

.c0:hover {
  background: #37cbaa;
}

<a
  className="c0"
  color="green"
  href="http://www.test.com"
  rel="noopenner noreferrer"
  target="_blank"
>
  <div>
    Click me
  </div>
</a>
`;
